base_classes/NXdata.nxdl.xml

```diff
@@ -85,124 +85,170 @@
+			  @axes = ["x", "y"]
 			  data: float[10,20]
 			  x: float[10]        --> coordinates along the first dimension
 			  y: float[20]        --> coordinates along the second dimension
 
 		In this example each data point ``data[i,j]`` has axis coordinates ``[x[i], y[j]]``.
 
-		However, the fields can also have a rank greater than 1, in which case the rank of each
-		:ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` must be equal to the number of data dimensions it spans.
+		Note that `@x_indices` and `@y_indices` attributes can be omitted in this case. However it is strongly encouraged to provide them.
+
+		`2. Multi-dimensional axes`
+
+		When coordinates do not form a regular grid, multi-dimensional :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` fields
+		can be used. Just remember that the shape of an :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` field must be equal to the shape
+		of the :ref:`data &lt;/NXdata/DATA-field&gt;` dimensions it spans.
 
-		An example of this would be
+		An example of this would be 2D scatter data where each data point has unique coordinates
 
 		.. code-block::
 
 			data:NXdata
 			  @signal = "data"
-			  @axes = ["x", "y"]  --> the order does NOT matter
+			  @axes = ["x", "y"]
 			  @x_indices = [0, 1]
 			  @y_indices = [0, 1]
 			  data: float[10,20]
 			  x: float[10,20]     --> coordinates along both dimensions
 			  y: float[10,20]     --> coordinates along both dimensions
 
-		In this example each data point ``data[i,j]`` has axis coordinates ``[x[i,j], y[i,j]]``.
-
-		`Dimensions`
-
-		The data dimensions annotated by an :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` field are defined by the
-		:ref:`AXISNAME_indices &lt;/NXdata@AXISNAME_indices-attribute&gt;` attribute. When this attribute is missing,
-		the position(s) of the :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` string in the
-		:ref:`axes &lt;/NXdata@axes-attribute&gt;` attribute are used.
+		In this example each data point ``data[i,j]`` has axis coordinates ``[x[i,j], y[i,j]]`` and when
+		plotting, `x` is used along the first data dimension by default and `y` along the second data dimension.
+		Since `x` and `y` span both dimensions, a reader could choose to use `y` for the first dimension
+		and `x` for the second as an alternative to the default. A writer could also choose to not specify
+		any default by specifying `@axes = [".", "."]` and leave the decision up the the reader.
 
-		When all :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` fields are one-dimensional, and none of the data dimensions
-		have more than one axis, the :ref:`AXISNAME_indices &lt;/NXdata@AXISNAME_indices-attribute&gt;` attributes
-		are often omitted. If one of the data dimensions has no :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` field,
-		the string “.” can be used in the corresponding index of the axes list.
+		Note that omitting `@x_indices` would result in `@x_indices = [0]` as derived from the position of `"x"` in `@axes`.
```

Comment:
I think it is better to say an `AXISNAME_indices` attribute must be provided for multi-dimensional `AXISNAME` fields.

Reply:
But technically it is still optional right?